### PR TITLE
Allow for non-binary species to use it/its pronouns 

### PR DESCRIPTION
**Pull request by seafork**
### Remove `TRAIT_AGENDER` from all ethereals

This is necessary as it's attachment to ethereals cause their genders to be overridden. This only removes it's attachment to ethereals and not the logic associated with it. I feel okay doing this as `TRAIT_AGENDER` has been (seemingly) eclipsed by `sexes`. This does however leave dead code.

### Allow for non-binary species to use it/its pronouns

Change the `tgui` gender selection to always show up; however, only prompt non-binary species with the choice of `they/them` and `it/its`. The server has also been changed to prevent overriding the user's choice while also preventing other pronouns being used. While this implementation does leave more to desire, it's the most unobtrusive way I could implement such a change.

## Why It's Good For The Game

inclusivity good!

## Changelog

:cl:
qol: allow for non-binary species (ethereals, plasmamen) to use it/its pronouns.
/:cl:

### CODE REVIEW

**Comment by itseasytosee** [comment]
It does seem to be a bit contrived to relate how non-sexed species would relate their preferred pronouns in relation to an undefined non-English future language  completely outside the masculine feminine spectrum, supposedly converted from their own preferences in their own personal non-descript ethereal language which would have a completely different or relatively absent set of rules for identification, and probably would not have pronouns as we know them at all, let alone any sense of gender euphoria. Maybe plasmamen because they were once human and thefore have some shred of gender identity within them left from the mind wipe? But that seems pretty contrived in its own right, and also gives the implication that somehow the dead body of a creature has a gender identity, but never one based on sex. 

This would probably be a viable complaint for someone say, trying to write a novel, or a movie. But the themeing of aliens in our game are pretty much like mass effect, star wars, star trek, and shitty star trek fan fiction all had an orgy so I doubt that literally anyone (including myself) would feel like the integrity of the worldbuilding in that aspect could be shattered even further.

Literally no one of sound mind would care that we are letting ethereal use it/that. But here I propose, why are we stopping it there?
If you are claiming that ethereals can have a gender identity or a sense of belonging, why the hell are we even limiting them to the two pronouns systems that aren't masculine/feminine based? If for whatever convoluted reason ethereals care about human derived gender identity, why would they not also sometimes choose to identify as he or her? 
I think it would be a weird design decision to be all like "he he, ethereals can have a little slice of human gender identity, as a treat." 
I kinda question your "why its good for the game message" because, who are we being inclusive to? If you say people with it/its pronouns, then are ethereals not already inherently exclusive to people who really want to play a character with he/her pronouns?
Because this is a roleplaying game with alien people who didn't develop as humans did, does it make sense to confine those people to the same pronoun standards as humans? Is it non-inclusive to not do this?
Its not like we are even trying to draw a parallel or make commentary about a situation by doing this.
And if we drop all that and say "we just want ethereals and human gender identities to be similar and any sort of worldbuilding implications are negligible or explainable by [Insert fanfiction about ethereals and human first contact here]," why are we even bothering limiting them to they/them and it/its?

**tldr; if we are going to bother with this we might as well just let them use whatever pronouns they want**

**Comment by BlueMemesauce** [comment]
> It does seem to be a bit contrived to relate how non-sexed species would relate their preferred pronouns in relation to an undefined non-English future language completely outside the masculine feminine spectrum, supposedly converted from their own preferences in their own personal non-descript ethereal language which would have a completely different or relatively absent set of rules for identification, and probably would not have pronouns as we know them at all, let alone any sense of gender euphoria. Maybe plasmamen because they were once human and thefore have some shred of gender identity within them left from the mind wipe? But that seems pretty contrived in its own right, and also gives the implication that somehow the dead body of a creature has a gender identity, but never one based on sex.
> 
> This would probably be a viable complaint for someone say, trying to write a novel, or a movie. But the themeing of aliens in our game are pretty much like mass effect, star wars, star trek, and shitty star trek fan fiction all had an orgy so I doubt that literally anyone (including myself) would feel like the integrity of the worldbuilding in that aspect could be shattered even further.
> 
> Literally no one of sound mind would care that we are letting ethereal use it/that. But here I propose, why are we stopping it there? If you are claiming that ethereals can have a gender identity or a sense of belonging, why the hell are we even limiting them to the two pronouns systems that aren't masculine/feminine based? If for whatever convoluted reason ethereals care about human derived gender identity, why would they not also sometimes choose to identify as he or her? I think it would be a weird design decision to be all like "he he, ethereals can have a little slice of human gender identity, as a treat." I kinda question your "why its good for the game message" because, who are we being inclusive to? If you say people with it/its pronouns, then are ethereals not already inherently exclusive to people who really want to play a character with he/her pronouns? Because this is a roleplaying game with alien people who didn't develop as humans did, does it make sense to confine those people to the same pronoun standards as humans? Is it non-inclusive to not do this? Its not like we are even trying to draw a parallel or make commentary about a situation by doing this. And if we drop all that and say "we just want ethereals and human gender identities to be similar and any sort of worldbuilding implications are negligible or explainable by [Insert fanfiction about ethereals and human first contact here]," why are we even bothering limiting them to they/them and it/its?
> 
> **tldr; if we are going to bother with this we might as well just let them use whatever pronouns they want**

plasmamen are chad asexuals and have no need for mortal concepts of gender

idk about ethereals

**Comment by Jackal-boop** [comment]
> It does seem to be a bit contrived to relate how non-sexed species would relate their preferred pronouns in relation to an undefined non-English future language completely outside the masculine feminine spectrum, supposedly converted from their own preferences in their own personal non-descript ethereal language which would have a completely different or relatively absent set of rules for identification, and probably would not have pronouns as we know them at all, let alone any sense of gender euphoria. Maybe plasmamen because they were once human and thefore have some shred of gender identity within them left from the mind wipe? But that seems pretty contrived in its own right, and also gives the implication that somehow the dead body of a creature has a gender identity, but never one based on sex.
> 
> This would probably be a viable complaint for someone say, trying to write a novel, or a movie. But the themeing of aliens in our game are pretty much like mass effect, star wars, star trek, and shitty star trek fan fiction all had an orgy so I doubt that literally anyone (including myself) would feel like the integrity of the worldbuilding in that aspect could be shattered even further.
> 
> Literally no one of sound mind would care that we are letting ethereal use it/that. But here I propose, why are we stopping it there? If you are claiming that ethereals can have a gender identity or a sense of belonging, why the hell are we even limiting them to the two pronouns systems that aren't masculine/feminine based? If for whatever convoluted reason ethereals care about human derived gender identity, why would they not also sometimes choose to identify as he or her? I think it would be a weird design decision to be all like "he he, ethereals can have a little slice of human gender identity, as a treat." I kinda question your "why its good for the game message" because, who are we being inclusive to? If you say people with it/its pronouns, then are ethereals not already inherently exclusive to people who really want to play a character with he/her pronouns? Because this is a roleplaying game with alien people who didn't develop as humans did, does it make sense to confine those people to the same pronoun standards as humans? Is it non-inclusive to not do this? Its not like we are even trying to draw a parallel or make commentary about a situation by doing this. And if we drop all that and say "we just want ethereals and human gender identities to be similar and any sort of worldbuilding implications are negligible or explainable by [Insert fanfiction about ethereals and human first contact here]," why are we even bothering limiting them to they/them and it/its?
> 
> **tldr; if we are going to bother with this we might as well just let them use whatever pronouns they want**

idk how well adding a custom pronoun field would go but adding it/its and neo pronouns would be pretty swag

**Comment by BlueMemesauce** [comment]
> idk how well adding a custom pronoun field would go but adding it/its and neo pronouns would be pretty swag

we already added it/its, thats literally the entire point of this pr.
we are not adding neopronouns

**Comment by Jackal-boop** [comment]
> > idk how well adding a custom pronoun field would go but adding it/its and neo pronouns would be pretty swag
> 
> we already added it/its, thats literally the entire point of this pr. we are not adding neopronouns

you are not the determiner of if we do or do not :3

**Comment by itseasytosee** [comment]
> > [itseasytosee](https://github.com/itseasytosee): wall of text
> 
> plasmamen are chad asexuals and have no need for mortal concepts of gender
> 
> idk about ethereals

Yeah homie thats my point.

**Comment by BlueMemesauce** [comment]
> > > [itseasytosee](https://github.com/itseasytosee): wall of text
> > 
> > 
> > plasmamen are chad asexuals and have no need for mortal concepts of gender
> > idk about ethereals
> 
> Yeah homie thats my point. It doesn't make sense for them to care, and if they do care then they would probably care about he/her as well.

Right but they could still have a difference between it/they for different reasons. It makes sense for them to use "it", since plasmamen are fungi and aren't animals. However, some plasmamen could dispute this and want to use "they" to seem more personlike. Additionally, a more unified plasmaman want to use "it" to emphasize that they are a single unit and not just a group of fungi.

**Comment by seafork** [comment]
> It does seem to be a bit contrived to relate how non-sexed species would relate their preferred pronouns in relation to an undefined non-English future language completely outside the masculine feminine spectrum

Right, I totally agree that letting players choose their gender identity regardless of their sex (or lack thereof) is the correct way forward. Not only on the grounds of inclusion, but also lore wise, as you've stated. Decoupling gender and sex inside the code would also (hopefully) have the benefit of making adding new pronouns easier (at the cost of the initial refactor, of course.)

> Literally no one of sound mind would care that we are letting ethereal use it/that. But here I propose, why are we stopping it there?

With this being my first commit, my intentions were to make a small a change as possible, both code wise and lore wise. (Something followed almost to a fault [see `tgui` code 🥲]) That being said, if the community comes to a consensus, I'd be more than happy to do the legwork and (try to) implement whatever.

(Though in the meantime, I don't see adding this hurting anything!)

**Comment by itseasytosee** [comment]
> Right but they could still have a difference between it/they for different reasons. It makes sense for them to use "it", since plasmamen are fungi and aren't animals. However, some plasmamen could dispute this and want to use "they" to seem more personlike. Additionally, a more unified plasmaman want to use "it" to emphasize that they are a single unit and not just a group of fungi.

I made reference to this in my first post, any time you want to make up a lore reason as to why a species what's to identify as x, y, or z, you could also say "they might also want to use traditional human pronouns like he/her" to identify themselves. Its really an arbitrary restriction.

**Comment by Jacquerel** [comment]
> > > idk how well adding a custom pronoun field would go but adding it/its and neo pronouns would be pretty swag
> > we already added it/its, thats literally the entire point of this pr. we are not adding neopronouns
> you are not the determiner of if we do or do not :3

No but the host is and he blocked that idea when we added it/its. It has also been previously stated to be a bad idea by headcoders for maintainability reasons.
He/She/They/It are all in-engine for the default atom pronoun field, adding a wrapper for more would add both technical _and_ administrative headaches which nobody really wants to deal with.

I'll be honest i am partial to the argument of "if you aren't leaving them at one set choice then it doesn't make sense to give them two options instead of four", if you are capable of having the concept of a gender identity you might choose to be a girl, but I also don't really care very much.

**Comment by BlueMemesauce** [comment]
I kind of agree with the argument that they should be able to use gendered pronouns in uncommon cases like if they want to be more like humans.
But consider that most players don't really care about the lore, and will probably use "he" or "her" pronouns at a much higher rate. This removes a unique part of the species

**Comment by itseasytosee** [comment]
I wish roleplay servers were more about character style guides, and less about restricting player action. That way we could write up a whole bit about how ethereal pick their pronouns based on whatever color they glow or something equally stupid and call that it a night.

**Comment by LemonInTheDark** [comment]
I think it is cool and good that ethereals and plasmemes have these unique aspects to them, and I would be a bit sad to see them shaved down. I realize it lets people be more fluid with how their character is, but restrictions like that help define the species, at least a little. 

Also no neo-pronouns aren't gonna happen unless we are able to define custom gender vars. Open dream my beloved that's gonna be a fun mso argument.

Also also I thought we already decoupled sex and gender. that's why bodytype is the way it is.

**Comment by Mothblocks** [comment]
Ethereals not allowing pronoun choice was an intentional choice by Floyd as a lore thing, we talked about it together in the past

**Comment by seafork** [comment]
> Also also I thought we already decoupled sex and gender. that's why bodytype is the way it is.

I'm going to be honest, I'm not quite sure what I meant by that. In my defense, it was late!

> Ethereals not allowing pronoun choice was an intentional choice by Floyd as a lore thing, we talked about it together in the past

Right, that's what I thought, but I don't see how adding another enby pronoun choice would step on that.

**Comment by seafork** [comment]
Kicking myself for how obvious a fix #78668 was. This fixes all regressions introduced with the initial change. Sorry for letting it sit there for so long!

**Comment by ChungusGamer666** [comment]
> > > idk how well adding a custom pronoun field would go but adding it/its and neo pronouns would be pretty swag
> > 
> > 
> > we already added it/its, thats literally the entire point of this pr. we are not adding neopronouns
> 
> you are not the determiner of if we do or do not :3

MrStonedOne has literally said this.

**Comment by ChungusGamer666** [comment]
I had made a PR removing the sexes variable for being redundant, and Oranges in his infinite wisdom closed all my open PRs at the time due to dumb bullshit.
Removing the agender trait seems to be the incorrect approach here, implementing some kind of restriction for people with the agender trait to only use neutral pronouns seems more reasonable.

**Comment by LemonInTheDark** [comment]
Go away bob

**Comment by ChungusGamer666** [comment]
nuh uh

**Comment by Jacquerel** [comment]
> I had made a PR removing the sexes variable for being redundant, and Oranges in his infinite wisdom closed all my open PRs at the time due to dumb bullshit. 

He closed them because you were gitbanned for long enough that they would have been closed by the stalebot. You can reopen them if you want.

**Comment by ChungusGamer666** [comment]
> He closed them because you were gitbanned for long enough that they would have been closed by the stalebot. You can reopen them if you want.

I won't, thank you.

**Comment by seafork** [comment]
Bumping this PR for review.

**Review by Ghommie — CHANGES_REQUESTED** [review]
I'm joining no argument, but since you've mentioned it, you should also remove the dead code related to `TRAIT_AGENDER`.
